web/packages/teleport/src/Integrations/status/AwsOidc/types.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Integration } from 'teleport/services/integrations';
+
+export type AwsOidcHeaderProps = {
```

Comment:
In general, types should be close to the code that uses them, especially when they are component props, imported in one place. 
IMO we overuse the `types.ts` files, usually there is no need to separate types from the implementation.

Here, the type can be even inlined:
```tsx
export function AwsOidcHeader({
  integration,
}: {
  integration: Integration | null;
}) {
```